client/src/bundles/UserForms.js

```diff
@@ -1,738 +1,742 @@
+    }
+
+    this.currentStep = step;
+    this.currentStep.show();
+
+    // Record the user has viewed the step.
+    this.currentStep.viewed = true;
+    this.currentStep.$el.addClass('viewed');
+  };
+
+  /**
+   * @func UserForm.jumpToStep
+   * @param {number} stepNumber
+   * @param {boolean} [direction] - Defaults to forward (true).
+   * @desc Jumps to a specific form step.
+   */
+  UserForm.prototype.jumpToStep = (stepNumber, direction) => {
+    const targetStep = this.steps[stepNumber];
+    let isValid = false;
+    const forward = direction === void 0 ? true : direction;
+
+    // Make sure the target step exists.
+    if (targetStep === void 0) {
+      return;
+    }
+
+    // Make sure the step we're trying to set as current is not
+    // hidden by custom display rules. If it is then jump to the next step.
+    if (targetStep.conditionallyHidden()) {
+      if (forward) {
+        this.jumpToStep(stepNumber + 1);
+      } else {
+        this.jumpToStep(stepNumber - 1);
+      }
+
+      return;
+    }
+
+    // Validate the form.
+    // This well effectivly validate the current step and not the entire form.
+    // This is because hidden fields are excluded from validation, and all fields
+    // on all other steps, are currently hidden.
+    isValid = this.$el.valid();
+
+    // Set the 'valid' property on the current step.
+    this.currentStep.valid = isValid;
+
+    // Users can navigate to step's they've already viewed even if the current step is invalid.
+    if (isValid === false && targetStep.viewed === false) {
+      return;
+    }
+
+    this.currentStep.hide();
+    this.setCurrentStep(targetStep);
+
+    this.$el.trigger('userform.form.changestep', [targetStep.id]);
+  };
+
+  /**
+   * @func UserForm.nextStep
+   * @desc Advances the form to the next step.
+   */
+  UserForm.prototype.nextStep = () => {
+    this.jumpToStep(this.steps.indexOf(this.currentStep) + 1, true);
+  };
+
+  /**
+   * @func UserForm.prevStep
+   * @desc Goes back one step (not bound to browser history).
+   */
+  UserForm.prototype.prevStep = () => {
+    this.jumpToStep(this.steps.indexOf(this.currentStep) - 1, false);
+  };
+
+  /**
+   * @func ErrorContainer
+   * @constructor
+   * @param {object} element - The error container element.
+   * @return {object} - The ErrorContainer instance.
+   * @desc Creates an error container. Used to display step error messages at the top.
+   */
+  function ErrorContainer(element) {
+    this.$el = element instanceof $ ? element : $(element);
+
+    // Set the error container's heading.
+    this.$el.find('h4').text(window.ss.i18n._t('UserForms.ERROR_CONTAINER_HEADER',
+      'Please correct the following errors and try again:'));
+
+    return this;
+  }
+
+  /**
+   * @func hasErrors
+   * @return boolean
+   * @desc Checks if the error container has any error messages.
+   */
+  ErrorContainer.prototype.hasErrors = () => {
+    return this.$el.find('.error-list').children().length > 0;
+  };
```

Comment:
this can be written as follows:
```
ErrorContainer.prototype.hasErrors = () => (
  this.$el.find('.error-list').children().length > 0
);
```
it's a common es6 practice, where `name = () => something;` and you get this as true: `name() === something`
the round brackets just makes it clear that it's wrapped to another line